controllers/bsl.go

```diff
@@ -252,6 +252,14 @@ func (r *DPAReconciler) updateBSLFromSpec(bsl *velerov1.BackupStorageLocation, d
 				bslSpec.Config["s3Url"] = s3Url
 			}
 		}
+
+		// Since the AWS SDK upgrade in velero-plugin-for-aws, data transfer to BSL bucket fails unless
+		// we specify the checksumAlgorithm. We will set it empty string if checksumAlgorithm is not specified by the user
+		// Setting it to an empty string will default to using the CRC32 algorithm for checksum calculation.
```

Comment:
setting it to an empty string will result in no checksum being calculated, right?
Reference https://github.com/vmware-tanzu/velero-plugin-for-aws/blob/main/backupstoragelocation.md?plain=1#L122

Reply:
@shubham-pampattiwar I think @mateusoliveira43 is right here. Especially since with IBM Cloud, we know that all four supported algorithms fail to work.

Reply:
"Setting it to an empty string will skip checksum calculation" -- that's what we need to say here. CRC32 will not be chosen if this is empty -- CRC32 is only chosen if the field is missing entirely (which is why IBM and Noobaa fail without this).

Reply:
```
		// Since the AWS SDK upgrade in velero-plugin-for-aws, data transfer to BSL bucket fails unless
		// if the chosen checksumAlgorithm doesn't work fo rthe provider. Velero sets this to CRC32 if not
		// chosen by the user. We will set it empty string if checksumAlgorithm is not specified by the user
		// to bypass checksum calculation entirely. If your s3 provider supports checksum calculation,
		// then you should specify this value in the config.
```

Reply:
Alright, got your thinking. Still believe doc is wrong

We have 3 scenarios, right?

- if user does NOT set the field AND has set `s3Url`, if will trigger and `checksumAlgorithm` will be empty string, which will result in no calculation

- if user does NOT set the field but also has NOT set `s3Url`, `checksumAlgorithm` will be `CRC32` (velero default)

- if user does set it, it will use what user specified (doc is saying it different, not?)

Reply:
@mateusoliveira43 We're simplifying this and avoiding the `s3Url` check so we'll always default to "", if users want CRC32, they'll need to set it.